Lib/Src/CJsonParser.cpp

```diff
@@ -140,15 +140,15 @@ void CJsonParser::processDigit(char current_char)
  */
 void CJsonParser::processDecimalPoint(char current_char)
 {
+    m_current_token.text.append(1, current_char);
```

Comment:
This one is a bit suspect, no? In one branch of `if()` if token is integer, character is appended then type is changed to float. That makes sense. In the other branch, however, the sequence changes. It used to be
```
finishCurrentToken();
//then new token begins...
m_curent_token.text.append(1, current_char);
m_curent_token.type = INVALID;
```
In old logic decimal point used to be attached to new token, but now it is attached to old token, which is then finished. But the new token is declared invalid. I think this is an error in logic.